### PR TITLE
Split physical_plan_tpch into separate benchmarks

### PR DESCRIPTION
**Pull request by simonvandel**
## Which issue does this PR close?

Relates to https://github.com/apache/arrow-datafusion/issues/8638

## Rationale for this change

The `physical_plan_tpch` benchmark was taking a long time to run, without any feedback on the performance of each query.

I think that splitting the benchmark up into a benchmark per query will allow better visibility into how code changes affect individual queries. 

## What changes are included in this PR?

Move `physical_plan_tpch` benchmark into multiple `physical_plan_tpch_{qn}` benchmarks.

## Are these changes tested?

Yes, I have verified the benchmarks run.

## Are there any user-facing changes?

No

### CODE REVIEW

**Comment by alamb** [comment]
The clippy issue was fixed on main, so I think if you merge up from main the CI will pass

File: datafusion/core/benches/sql_planner.rs [hunk 1]
```diff
@@ -224,53 +224,17 @@ fn criterion_benchmark(c: &mut Criterion) {
...
-        b.iter(|| physical_plan(&ctx, &q17_sql));
-        b.iter(|| physical_plan(&ctx, &q18_sql));
-        b.iter(|| physical_plan(&ctx, &q19_sql));
-        b.iter(|| physical_plan(&ctx, &q20_sql));
-        b.iter(|| physical_plan(&ctx, &q21_sql));
-        b.iter(|| physical_plan(&ctx, &q22_sql));
-    });
+    for q in [
+        "q1", "q2", "q3", "q4", "q5", "q6", "q7", "q8", "q9", "q10", "q11", "q12", "q13",
+        "q14", // "q15",
```
[the remark this answers is not in the document]
Comment (matthewmturner) [reply datafusion/core/benches/sql_planner.rs:229]: I believe that query didnt work when i ran it and then i saw somewhere else that it was commented out which is why i did. but now i'm struggling to find where else that query was commented out.  If the query works then i think ok to uncomment.

[the remark this answers is not in the document]
Comment (simonvandel) [reply datafusion/core/benches/sql_planner.rs]: Added a comment on why it is not run in the benchmark

**Review by alamb — APPROVED** [review]
Thank you @simonvandel

**Comment by alamb** [comment]
Thanks again @simonvandel

**Pull request merged**